experimental/segmenter/build.rs

```diff
@@ -0,0 +1,404 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu::properties::{maps, sets, WordBreak};
+use icu_provider_fs::FsDataProvider;
+use serde::Deserialize;
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+#[derive(Deserialize, Debug)]
+struct SegmenterPropertyValueMap {
+    codepoint: Option<Vec<u32>>,
+    left: Option<String>,
+    right: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct SegmenterProperty {
+    name: String,
+    value: Option<SegmenterPropertyValueMap>,
+}
+
+#[derive(Deserialize, Debug)]
+struct SegmenterState {
+    left: Vec<String>,
+    right: Vec<String>,
+    break_state: Option<bool>,
+}
+
+#[derive(Deserialize, Debug)]
+struct SegmenterRuleTable {
+    segmenter_type: String,
+    tables: Vec<SegmenterProperty>,
+    rules: Vec<SegmenterState>,
+}
```

Comment:
suggestion: Could you add some documentation to these struct so that it is easier to understand the `word.json`?